doc/grpc_client/design_doc.md

```diff
@@ -0,0 +1,509 @@
+## gRPC client for active-actve DualToR scenario design
+
+
+Table of Contents
+=================
+* [Scope](#scope)
+* [Requirements](#Requirements)
+* [why gRPC](#whygrpc)
+* [HardWare Overview and Overall Architecture](#hardware-overview-and-overall-archtecture)
+  * [Hardware Overview](#hardware-overview)
+  * [Host Architecture](#host-architecture)
+  * [DualToR architecture](#dualtor-redundancy-achievment-using-active-active-solution)
+* [Proto and Schema Definition](#proto-and-schema-definition)
+  * [Proto Definition for forwarding State](#proto-definition-interface-to-state-machine-for-getset-admin-state-of-the-fpga-ports)
+  * [Schema Definition for DB's ](#ycabled-functional-schema-for-data-exchanged-between-orchagent-and-linkmgr)
+* [gRPC channel customisation and Telemetry Schema](#grpc-channel-customisation-and-telemetry-schema)
+  * [Keepalive mechanism for channels](#keepalive-for-grpc-channelstub)
+* [gRPC client communicate to SoC over Loopback IP](#grpc-client-communicate-to-soc-over-loopback-ip)
+* [gRPC commuication over secure channel](#grpc-commuication-over-secure-channel)
+* [gRPC client initialization/deployment](#deployment)
+* [gRPC commuication to NIC simulator](#grpc-communication-with-nic-simulator)
+* [Proto and Schema Definition for Aync notificaion](#proto-definition-interface-to-soc-to-notify-ycabled-about-service-notification)
+
+
+## Revision
+
+| Rev | Date     | Author          | Change Description |
+|:---:|:--------:|:---------------:|--------------------|
+| 0.1 | 04/1/22 | Vaibhav Dahiya  | Initial version    |
+| 0.2 | 02/1/22 | Vaibhav Dahiya  | Make chnages to be shared to Core Team    |
+
+## Scope
+gRPC client design doc which would communicate with the SoC in DualToR Active-active setup/ and Nic-simulator for testing infrastructure in SONiC MGMT.
+### Overview
+
+This document summarizes the approach taken to accommodate gRPC client
+for DualToR active active scenario. The gRPC client daemon's
+main purpose is to provide a way for linkmgr to exchange RPC's with SoC
+and do this within SONiC PMON docker
+
+
+## Requirements
+
+- provide a service/daemon in SONiC to run in DualToR mode, which can interact with Platform API as well interact with state machine(aka Linkmgr) and orchagent to provide capability for it to get/set Link State/Forwarding State etc. from SoC(gRPC server listening to the client)

```

Comment:
isn't gRPC service running on PMON is stateless?

Reply:
its stateless